price-feeder/oracle/provider/mexc.go

```diff
@@ -0,0 +1,504 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gorilla/websocket"
+	"github.com/rs/zerolog"
+	"github.com/umee-network/umee/price-feeder/oracle/types"
+)
+
+const (
+	mexcWSHost   = "wbs.mexc.com"
+	mexcWSPath   = "/raw/ws"
+	mexcRestHost = "https://www.mexc.com"
+	mexcRestPath = "/open/api/v2/market/ticker"
+)
+
+var _ Provider = (*MexcProvider)(nil)
+
+type (
```

Comment:
instead of creating a `type` block, we can add type to every definition. This way we avoid having one big indentation.